src/containers/filedataset.jl

```diff
@@ -0,0 +1,57 @@
+matches(re::Regex) = f -> matches(re, f)
+matches(re::Regex, f) = !isnothing(match(re, f))
+const RE_IMAGEFILE = r".*\.(gif|jpe?g|tiff?|png|webp|bmp)$"i
+isimagefile(f) = matches(RE_IMAGEFILE, f)
+
+"""
+    rglob(filepattern, dir = pwd(), depth = 4)
+
+Recursive glob up to `depth` layers deep within `dir`.
+"""
+function rglob(filepattern = "*", dir = pwd(), depth = 4)
+    patterns = [repeat("*/", i) * filepattern for i in 0:(depth - 1)]
+
+    return vcat([glob(pattern, dir) for pattern in patterns]...)
+end
+
+"""
+    loadfile(file)
+
+Load a file from disk into the appropriate format.
+"""
+function loadfile(file::String)
+    if isimagefile(file)
+        # faster image loading
+        return FileIO.load(file, view = true)
```

Comment:
I think this special case is no longer necessary with ImageIO.jl now defaulting to JpegTurbo.jl (`view` is ImageMagick specific I think.